mlrun/api/api/deps.py

```diff
@@ -70,20 +70,34 @@ def verify_api_state(request: Request):
             "memory-reports",
         ]
         if not any(enabled_endpoint in path for enabled_endpoint in enabled_endpoints):
+
+            # waiting for migration
             message = (
                 "API is waiting for migrations to be triggered. Send POST request to /api/operations/migrations to"
                 " trigger it"
             )
+
+            # migration in progress
             if (
                 mlrun.mlconf.httpdb.state
                 == mlrun.api.schemas.APIStates.migrations_in_progress
             ):
                 message = "Migrations are in progress"
+
+            # migration failed
             elif (
                 mlrun.mlconf.httpdb.state
                 == mlrun.api.schemas.APIStates.migrations_failed
             ):
                 message = "Migrations failed, API can't be started"
+
+            # waiting for chief
+            elif (
+                mlrun.mlconf.httpdb.state
+                == mlrun.api.schemas.APIStates.waiting_for_chief
+            ):
+                message = "API is waiting for chief to be ready."
+
             raise mlrun.errors.MLRunPreconditionFailedError(message)
```

Comment:
What do you think of this? Or even maybe put this dictionary in a `get_precondition_failed_message` method in the `APIStates` class.
```suggestion
            precondition_failed_messages = {
                mlrun.api.schemas.APIStates.waiting_for_migration: "API is waiting for migrations to be triggered. Send POST request to /api/operations/migrations to trigger it",
                mlrun.api.schemas.APIStates.migrations_in_progress: "Migrations are in progress",
                mlrun.api.schemas.APIStates.migrations_failed: "Migrations failed, API can't be started",
                mlrun.api.schemas.APIStates.waiting_for_chief: "API is waiting for chief to be ready.",
            }

            raise mlrun.errors.MLRunPreconditionFailedError(precondition_failed_messages.get(mlrun.mlconf.httpdb.state))
```

Reply:
will do, great idea